source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -147,18 +152,19 @@ SplitRequestPtr SimpleRequest::create(Router& router,
   std::unique_ptr<SimpleRequest> request_ptr{
       new SimpleRequest(callbacks, command_stats, time_source, latency_in_micros)};
 
-  auto conn_pool = router.upstreamPool(incoming_request->asArray()[1].asString());
-  if (conn_pool) {
-    request_ptr->conn_pool_ = conn_pool;
-    request_ptr->handle_ = conn_pool->makeRequest(incoming_request->asArray()[1].asString(),
-                                                  *incoming_request, *request_ptr);
+  std::string& key = incoming_request->asArray()[1].asString();
```

Comment:
const, same elsewhere in this diff.

Reply:
It's possible that the key be changed by the subsequent calls to remove the prefix.

Reply:
OK thanks. I realize it's technically correct, but can you remove the reference and just do a copy? The compiler will optimize as appropriate. Same elsewhere.